### PR TITLE
Add Documentation for Making Dictionaries

### LINKED ISSUE #635: Yomitan dictionary making process documentation
There might be interest in official documentation for the dictionary making process.
There is [Yomitan Import](https://github.com/themoeway/yomitan-import), but it only supports a limited format and is not powerful enough. There is no official documentation on the supported tag categories https://github.com/themoeway/yomitan/blob/48f1d012ad5045319d4e492dfbefa39da92817b2/ext/css/display.css#L761-L790.
Links to projects such as https://github.com/stephenmk/jitenbot or https://github.com/MarvNC/yomichan-dict-builder/ might be useful as well.

### PR DESCRIPTION
**Pull request by MarvNC**
Resolves #635 

Would like some feedback on the overall structure and stuff to make it fit in better here. For now, these are just my own docs rewritten to be third person.

### CODE REVIEW

File: docs/making-yomitan-dictionaries.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+## Tools
+
+- [Yomichan Dictionary Builder](https://github.com/MarvNC/yomichan-dict-builder/): A node package that simplifies the process of making dictionaries, particularly useful for those using TypeScript or JavaScript.
+- [hasUTF16SurrogatePairAt](https://www.npmjs.com/package/@stdlib/assert-has-utf16-surrogate-pair-at): Important for checking if a kanji/hanzi is a surrogate pair, which affects string operations in JavaScript.
+- [japanese-furigana-normalize](https://github.com/MarvNC/japanese-furigana-normalize): A utility function to normalize Japanese readings containing furigana, ensuring proper alignment with kanji characters.
+
+## Read the Schemas
+
+Familiarity with the [Yomitan schemas](https://github.com/themoeway/yomitan/tree/master/ext/data/schemas) is essential. These schemas define the structure of Yomichan dictionaries. Helpful resources for interpreting JSON schemas include [codebeautify](https://codebeautify.org/jsonviewer/), [json-schema-viewer](https://json-schema-viewer.vercel.app/), and [jsonhero](https://jsonhero.io/).
```
Comment (Kuuuube) [inline docs/making-yomitan-dictionaries.md:34]:
```suggestion
Familiarity with the [Yomitan schemas](https://github.com/themoeway/yomitan/tree/master/ext/data/schemas) is essential. These schemas define the structure of Yomitan dictionaries. Helpful resources for interpreting JSON schemas include [codebeautify](https://codebeautify.org/jsonviewer/), [json-schema-viewer](https://json-schema-viewer.vercel.app/), and [jsonhero](https://jsonhero.io/).
```

File: docs/making-yomitan-dictionaries.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+Below is a list of Yomichan dictionary schemas, their expected filenames, and their usage:
```
Comment (Kuuuube) [inline docs/making-yomitan-dictionaries.md:36]:
```suggestion
Below is a list of Yomitan dictionary schemas, their expected filenames, and their usage:
```

File: docs/making-yomitan-dictionaries.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    {
+      "fileMatch": ["term_meta_bank_*.json"],
+      "url": "https://github.com/themoeway/yomitan/raw/master/ext/data/schemas/dictionary-term-meta-bank-v3-schema.json"
+    }
+  ],
+```
+
+## Conjugation
+
+For Yomichan to conjugate Japanese terms, they need the appropriate part of speech tag. The part of speech labels are documented on the [official JMDict page](http://www.edrdg.org/jmdictdb/cgi-bin/edhelp.py?svc=jmdict&sid=#kw_pos). An [npm package](https://www.npmjs.com/package/yomichan-dict-reader) can help with extracting conjugations from JMDict, as demonstrated in the [JP-Mongolian dictionary](https://github.com/MarvNC/yomichan-dictionaries/blob/master/mongolian/scrape.js).
```
Comment (Kuuuube) [inline docs/making-yomitan-dictionaries.md:88]:
```suggestion
For Yomitan to conjugate Japanese terms, they need the appropriate part of speech tag. The part of speech labels are documented on the [official JMDict page](http://www.edrdg.org/jmdictdb/cgi-bin/edhelp.py?svc=jmdict&sid=#kw_pos). An [npm package](https://www.npmjs.com/package/yomichan-dict-reader) can help with extracting conjugations from JMDict, as demonstrated in the [JP-Mongolian dictionary](https://github.com/MarvNC/yomichan-dictionaries/blob/master/mongolian/scrape.js).
```

**Review by Kuuuube — CHANGES_REQUESTED** [review]
Should remove references to Yomichan (besides external libraries).

File: docs/making-yomitan-dictionaries.md [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+    {
+      "fileMatch": ["term_meta_bank_*.json"],
+      "url": "https://github.com/themoeway/yomitan/raw/master/ext/data/schemas/dictionary-term-meta-bank-v3-schema.json"
+    }
+  ],
+```
+
+## Conjugation
+
+For Yomitan to conjugate Japanese terms, they need the appropriate part of speech tag. The part of speech labels are documented on the [official JMDict page](http://www.edrdg.org/jmdictdb/cgi-bin/edhelp.py?svc=jmdict&sid=#kw_pos). An [npm package](https://www.npmjs.com/package/yomichan-dict-reader) can help with extracting conjugations from JMDict, as demonstrated in the [JP-Mongolian dictionary](https://github.com/MarvNC/yomichan-dictionaries/blob/master/mongolian/scrape.js).
```
Comment (cashewnuttynuts) [inline docs/making-yomitan-dictionaries.md:88]: The part of speech label for other languages can be found in `ext/js/language/{language}/{language}-transforms.js` in the `conditions` part not prefixed with "Intermediate"

File: docs/making-yomitan-dictionaries.md [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+- [Conjugation](#conjugation)
+- [Tag Categories](#tag-categories)
+
+## Prerequisites
+
+A basic familiarity with JSON and some coding experience in a language of your choice is required.
+
+## General Process
+
+1. Acquire data (from a website, app, dump, etc.)
```
Comment (cashewnuttynuts) [inline docs/making-yomitan-dictionaries.md:21]: "Acquire dictionary data" should be more clear

File: docs/making-yomitan-dictionaries.md [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+2. Parse the data to make sense of it
```
Comment (cashewnuttynuts) [inline docs/making-yomitan-dictionaries.md:22]: "to make sense of it" isnt really needed imo

File: docs/making-yomitan-dictionaries.md (hunk 5 above)
Comment (MarvNC) [reply docs/making-yomitan-dictionaries.md:21]: How so? It's pretty obvious imo.

File: docs/making-yomitan-dictionaries.md (hunk 6 above)
Comment (MarvNC) [reply docs/making-yomitan-dictionaries.md:22]: I added this section a while ago after someone asked (imo) pretty obvious questions but I'm thinking maybe it's just not necessary then.

**Review by cashewnuttynuts — DISMISSED** [review]

File: docs/making-yomitan-dictionaries.md
Comment (jamesmaa) [inline docs/making-yomitan-dictionaries.md:1]: Maybe add a paragraph about where to get help if you need it (#yomitan-dictionaries). I also wonder if we should instructions to submit a issue/PR to Marv's dictionary collection when they publish it. Knowing how to "publish" your dictionaries would be helpful to increasing spread of dictionaries

**Review by jamesmaa — COMMENTED** [review]
Overall seems good

**Review by jamesmaa — APPROVED** [review]

**Pull request merged**